### PR TITLE
Implementation in Java #1999

### PR DESCRIPTION
**Pull request by Amit366**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

This is the last part of implementation in Java #1999

### CODE REVIEW

**Review by atarax665 — COMMENTED** [review]
The code looks copied from Geeks For Geeks.

**Comment by Amit366** [comment]
@atarax665 sir check it now.

**Comment by Amit366** [comment]
And also please add the tags for this merged PR.
https://github.com/TesseractCoding/NeoAlgo/pull/2058

File: Java/cp/Longest_Pallindromicsubstring.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ 
+public static void main(String[] args)
+{
+    Scanner sc=new Scanner(System.in);
+    System.out.println("Enter word");
+    String str = sc.nextLine();
+    System.out.print("\nLength of longest pallindrome substring is: "
+         + longestPallindromeSubstr(str));
+}
+}
```
Comment (HarshCasper) [inline Java/cp/Longest_Pallindromicsubstring.java]: Add sample I/O and a new line at the EOF

File: Java/cp/Longest_Pallindromicsubstring.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// Java solution for longest palindrome in a substring
```
Comment (HarshCasper) [inline Java/cp/Longest_Pallindromicsubstring.java]: Add a beginner pitch about your Code/Algorithm

File: Java/cp/Longest_Pallindromicsubstring.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+// Function to print longest subString
+static void printSubStr(String str, int low, int high)
+{
+    for (int i = low; i <= high; ++i)
+        System.out.print(str.charAt(i));
+}
+
+static int longestPallindromeSubstr(String str)
+{
+    // get length of input String
```
Comment (HarshCasper) [inline Java/cp/Longest_Pallindromicsubstring.java]: Isn't the comment kinda redundant? 

image

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by HarshCasper** [comment]
> @atarax665 sir check it now.

Do you actually mean that you picked code from GFG and edited it later? If yes, we would have to close the PR on grounds of plagiarism

**Comment by Amit366** [comment]
@HarshCasper sir if you feel it is a copy of GFG then you can compare it.

**Comment by HarshCasper** [comment]
Not all my reviews are taken care of. Take a look at them and resolve the same.

**Comment by Amit366** [comment]
@HarshCasper sir what should be the beginning pitcher??

**Comment by atarax665** [comment]
> @HarshCasper sir what should be the beginning pitcher??

Beginner Pitch includes introduction of the problem that your code solves and description of your approach to solve it.

**Comment by Amit366** [comment]
@HarshCasper sir now is it fine??

File: Java/cp/Longest_Pallindromicsubstring.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
+
+// This is a Java program  to find the longest palindromicsubstring from a word using dynamic programming approach
```
Comment (atarax665) [inline Java/cp/Longest_Pallindromicsubstring.java]: Go through some merged PR to get a better idea of how to write a beginner pitch of a program.

**Comment by Amit366** [comment]
@atarax665 I have gone through some of the merged PRs. In some there are no beginner pitch and in others it is written similar to mine.

**Comment by Amit366** [comment]
And already my 3 PRs have been merged and  in them I have written in this manner then what's the problem in this??

**Comment by atarax665** [comment]
> And already my 3 PRs have been merged and in them I have written in this manner then what's the problem in this??

Okay, atleast explain what is a palindromic string with example and then this PR will be good to go.

**Comment by Amit366** [comment]
@atarax665 please review now

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**